### PR TITLE
ISPN-12622 Expose Cross-Site Views via CLI and REST endpoint

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-12622

### CODE REVIEW

**Comment by pruivo** [comment]
ps. I forgot to update the docs with the new commands :/

File: cli/src/main/resources/help/site.adoc [hunk 1]
```diff
@@ -62,3 +66,9 @@ Cancels the operation to receive state from `NYC`.
 
 `site clear-push-state-status --cache=myCache` +
 Clears the status of the push state operation for `mycache`.
+
+`site name` +
+Returns the local site name. If cross-site replication isn't available, it returns the word "local".
```
Comment (oraNod) [inline cli/src/main/resources/help/site.adoc:71]:
```suggestion
Returns the name of the local site. If cross-site replication is not configured, the name of the local site is always "local".
```

File: cli/src/main/resources/help/site.adoc [hunk 2]
```diff
@@ -62,3 +66,9 @@ Cancels the operation to receive state from `NYC`.
...
+
+`site view` +
+Returns a list with all the site's name, or an empty list ("[]") if cross-site replication isn't available.
```
Comment (oraNod) [inline cli/src/main/resources/help/site.adoc:74]:
```suggestion
Returns a list of names for all sites or an empty list ("[]") if cross-site replication is not configured.
```

**Comment by pruivo** [comment]
updated

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 3]
```diff
@@ -30,6 +30,10 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
 * `cluster_members` and `cluster_members_physical_addresses` an array of logical and physical addresses of the members of the cluster
 * `cluster_size` number of members in the cluster
 * `defined_caches` A list of all caches defined in the cache manager, excluding private caches but including internal caches that are accessible
+* `local_site` The local site name.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:33]:
```suggestion
* `local_site` The name of the local site. +
```

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 4]
```diff
@@ -30,6 +30,10 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
...
+If cross-site replication isn't configured, it returns "local".
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:34]:
```suggestion
If cross-site replication is not configured, {brandname} returns "local".
```

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 5]
```diff
@@ -30,6 +30,10 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
...
+* `sites_view` The cross-site replication view.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:35]:
```suggestion
* `sites_view` The list of sites that participate in cross-site replication. +
```

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 6]
```diff
@@ -30,6 +30,10 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
...
+If not configured, it returns an empty list.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:36]:
```suggestion
If cross-site replication is not configured, {brandname} returns an empty list.
```

**Comment by tristantarrant** [comment]
LGTM

**Comment by tristantarrant** [comment]
Odd, XSiteCliIT failed, but it works locally

**Comment by pruivo** [comment]
let me rebase on top of master to pick the fix for the container image. maybe it is related?

**Comment by pruivo** [comment]
hmmm. I've noticed the test is running twice
```
10:20:27.393 [INFO] Running org.infinispan.server.cli.XSiteCliIT
[OK:   239, KO:     0, SKIP:     0] Test starting: XSiteCliIT.testSiteView
[OK:   240, KO:     0, SKIP:     0] Test succeeded: XSiteCliIT.testSiteView
10:20:30.782 [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.384 s - in org.infinispan.server.cli.XSiteCliIT
...
10:29:47.768 [INFO] Running org.infinispan.server.cli.XSiteCliIT
[OK:   362, KO:     0, SKIP:     0] Test starting: XSiteCliIT.testSiteView
[OK:   362, KO:     1, SKIP:     0] Test failed: XSiteCliIT.testSiteView
10:30:31.220 [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 43.445 s <<< FAILURE! - in org.infinispan.server.cli.XSiteCliIT
```

does it make any sense to you?

**Comment by pruivo** [comment]
changing the test name worked :)

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**